# 10 - Beginner Exercises 
*   format()
*   F-String
* sep= & end= KeywordArgument
---



## ☕☕☕
1.Using the **.format()** method, write code that takes the user's name and then displays the message **```"Welcome to this program, user's name."```** 

In [1]:
# Write your own code in this cell
# Get user's name from user
user_name = input("What is your name?: ")
# Print Welcome message to user with user's name
print("Welcome to this program, {0}.".format(user_name))


Welcome to this program, Sina.


## ☕☕☕
2.Write code similar to the previous exercise, but this time use **```F-String```**. 
which one is faster?

In [2]:
# Write your own code in this cell
# Get user's name from user
user_name = input("What is your name?: ")
# Print Welcome message to user with user's name
print(f"Welcome to this program, {user_name}.")


Welcome to this program, Sina.


In [3]:
# Let's check which method is faster .format() or fstring
from time import time
user_name = "Sina"
# .format() execution time calculation
start_time_format = time()
print("Welcome to this program, {0}.".format(user_name))
print("--- %s seconds ---" % (time() - start_time_format))
# fstring execution time calculation
start_time_fstring = time()
print(f"Welcome to this program, {user_name}.")
print("--- %s seconds ---" % (time() - start_time_fstring))


Welcome to this program, Sina.
--- 0.0006825923919677734 seconds ---
Welcome to this program, Sina.
--- 0.0014221668243408203 seconds ---


##☕☕☕

3.Write a code that uses the F-String to display the information in my Tupel descriptively.

e.g.
**Celine Marie Claudette Dion was born in Canada and currently, she is 53 years old.**

In [4]:
# Write your own code in this cell
mytuple = ("David Jon Gilmour", 175, "United Kingdom")
# Prit Data from tuple into a string
print(f"{mytuple[0]} is from {mytuple[2]} and his height is {mytuple[1]} cm.")
       

David Jon Gilmour is from United Kingdom and his height is 175 cm.


## ☕☕
4. Using $sep=$ Keyword , print the information in the list below and separate the items with an **```"-~-"```**.  

In [5]:
# Write your own code in this cell
singers = ["Paul David Hewson", "Saeed Mohammadi" ,  "Aryana Sayeed"]
# Print Singers name in one line and seperate them with -~-
print(singers[0], singers[1], singers[2], sep="-~-")

       

Paul David Hewson-~-Saeed Mohammadi-~-Aryana Sayeed


## ☕☕
5.Print the following Tupel information for each item separately. Then repeat the same thing again, but using $end= $ Keyword, put the **```"---"```** at the end of each item. Then interpret the results. 

In [6]:
# Write your own code in this cell
C = ("Céline Marie Claudette Dion", 53, "Canada")
#Do not use  end= Keyword
print(C[0])
print(C[1])
print(C[2])
#Use end= Keyword
print(C[0], end="---")
print(C[1], end="---")
print(C[2], end="---")
       

Céline Marie Claudette Dion
53
Canada
Céline Marie Claudette Dion---53---Canada---

## ☕☕

6.Using **```positional_arguments```** insert the information into  **```mytext```**, then print it.  

In [7]:
# Write your own code in this cell
mytuple = ("Darroudi", "Surrealist", "Miami", "1958", "Florida State Art Center")

mytext= "{0}’s paintings are defined as {1} works. She was able to combine\n\
the delicacy and romantic spirit of Persian paintings with the elements of \n\
Surrealism.Darroudi's first solo exhibition was held in {2}, Miami-Dade County, \n\
Florida, in {3} at the invitation of the {4}."

print(mytext.format(*mytuple))

Darroudi’s paintings are defined as Surrealist works. She was able to combine
the delicacy and romantic spirit of Persian paintings with the elements of 
Surrealism.Darroudi's first solo exhibition was held in Miami, Miami-Dade County, 
Florida, in 1958 at the invitation of the Florida State Art Center.


**Can you explain why we used ```*mytuple``` instead of ```mytuple``` above?** 

## ☕☕

7.Using **```keyword_arguments```** complete the following phrase.

In [8]:
# Write your own code in this cell
print("The area of a circle with radius {r} is equal to {area}".format(r=5, area=5*5*3.14))

The area of a circle with radius 5 is equal to 78.5


## ☕

9.You are given the number $π$ to a precision of 20 decimal points.
Please print these numbers with two decimal ponit.

In [9]:
# Write your own code in this cell
pi = 3.14159265358979323846
print(f"{pi:.2f}")


3.14


## ☕

10.Please print the numbers below in this format. 
$$3.140000$$
$$1.618000$$
$$6.626068$$

In [10]:
# Write your own code in this cell
N = [3.14,1.618,6.626068]
print(f"{N[0]:.6f}")
print(f"{N[1]:.6f}")
print(f"{N[2]:.6f}")


3.140000
1.618000
6.626068


## 🌶️

11.Please print the numbers below in this format. 

005,421,241

083,992,949

331,002,651

In [11]:
# Write your own code in this cell
N=[5421241,83992949,331002651]
print(f'{f"{N[0]:,d}".zfill(11)}')
print(f'{f"{N[1]:,d}".zfill(11)}')
print(f'{f"{N[2]:,d}".zfill(11)}')

005,421,241
083,992,949
331,002,651
